assets/src/models/notificationData.ts

```diff
@@ -22,37 +37,88 @@ export const NotificationData = type({
     "adjusted",
     "operator_error",
     "traffic",
-    "chelsea_st_bridge_raised",
-    "chelsea_st_bridge_lowered",
   ]),
+  created_at: dateFromSeconds,
   route_ids: array(string()),
   run_ids: array(string()),
   trip_ids: array(string()),
   operator_name: nullable(string()),
   operator_id: nullable(string()),
   route_id_at_creation: nullable(string()),
-  start_time: number(),
-  end_time: nullable(number()),
+  start_time: dateFromSeconds,
+  end_time: nullable(dateFromSeconds),
+})
+
+export type BlockWaiverNotificationData = Infer<
+  typeof BlockWaiverNotificationData
+>
+
+export const BridgeNotificationData = union([
+  type({
+    $type: literal(NotificationType.BridgeMovement),
+    status: literal("raised"),
+    lowering_time: dateFromSeconds,
+  }),
+  type({
+    $type: literal(NotificationType.BridgeMovement),
+    status: literal("lowered"),
+  }),
+])
+
+export const NotificationData = type({
+  id: coerce(string(), number(), (i) => i.toString()),
+  created_at: dateFromSeconds,
   state: enums(["unread", "read", "deleted"]),
+  // Requires a field named `$type` to be present as the discriminator
+  content: union([BridgeNotificationData, BlockWaiverNotificationData]),
 })
 export type NotificationData = Infer<typeof NotificationData>
 
 export const notificationFromData = (
   notificationData: NotificationData
-): Notification => ({
-  id: notificationData.id.toString(),
-  createdAt: dateFromEpochSeconds(notificationData.created_at),
-  reason: notificationData.reason,
-  routeIds: notificationData.route_ids,
-  runIds: notificationData.run_ids,
-  tripIds: notificationData.trip_ids,
-  operatorName: notificationData.operator_name,
-  operatorId: notificationData.operator_id,
-  routeIdAtCreation: notificationData.route_id_at_creation,
-  startTime: new Date(notificationData.start_time * 1000),
-  state: notificationData.state,
-  endTime:
-    notificationData.end_time === null
-      ? null
-      : new Date(notificationData.end_time * 1000),
-})
+): Notification => {
+  let content: NotificationContentTypes
+
+  switch (notificationData.content.$type) {
+    case NotificationType.BlockWaiver: {
+      content = {
+        $type: NotificationType.BlockWaiver,
+        reason: notificationData.content.reason,
+        createdAt: notificationData.content.created_at,
```

Comment:
Maybe my prior comment about `created_at` is incorrect because I see here there is a difference between `notificationData.created_at` and `notificationData.content.created_at`, but I'm not sure why a BlockWaiver would have an additional `created_at` value from the db in its content

Reply:
Yeah me either :/